EtlPipelineBlobTrigger/course_docs.py

```diff
@@ -0,0 +1,275 @@
+    entry = {}
+    if key in lookup_table_raw_xml:
+        code = lookup_table_raw_xml[key]
+        entry['code'] = code
+        entry['label'] = lookup_table_local[code]
+    return entry
+
+
+def build_qualification(lookup_table_raw_xml, kisaims):
+    entry = {}
+    if 'KISAIMCODE' in lookup_table_raw_xml:
+        code = lookup_table_raw_xml['KISAIMCODE']
+        label = kisaims.get_kisaim_label_for_key(code)
+        entry['code'] = code
+        if label:
+            entry['label'] = label
+    return entry
+
+
+def build_course_entry(locations, locids, raw_inst_data, raw_course_data, kisaims):
+    outer_wrapper = {}
+    outer_wrapper['id'] = get_uuid()
+    outer_wrapper['created_at'] = datetime.datetime.utcnow().isoformat()
+    outer_wrapper['version'] = '1'
+
+    course = {}
+    if 'UKPRNAPPLY' in raw_course_data:
+        course['application_provider'] = raw_course_data['UKPRNAPPLY']
+    country = build_country(raw_inst_data)
+    if country:
+        course['country'] = country
+    distance_learning = build_code_label_entry(
+        raw_course_data, lookup.distance_learning_lookup, 'DISTANCE')
+    if distance_learning:
+        course['distance_learning'] = distance_learning
+    foundataion_year = build_code_label_entry(
+        raw_course_data, lookup.foundation_year_availability, 'FOUNDATION')
+    if foundataion_year:
+        course['foundation_year_availability'] = foundataion_year
+    if 'HONOURS' in raw_course_data:
+        course['honours_award_provision'] = raw_course_data['HONOURS']
+    course['institution'] = build_institution(raw_inst_data)
+    course['kis_course_id'] = raw_course_data['KISCOURSEID']
+    length_of_course = build_code_label_entry(
+        raw_course_data, lookup.length_of_course, 'NUMSTAGE')
+    if length_of_course:
+        course['length_of_course'] = length_of_course
+    links = build_links(locations, locids, raw_inst_data, raw_course_data)
+    if links:
+        course['links'] = links
+    location_items = build_location_items(
+        locations, locids, raw_inst_data, raw_course_data)
+    if location_items:
+        course['locations'] = location_items
+    mode = build_code_label_entry(raw_course_data, lookup.mode, 'KISMODE')
+    if mode:
+        course['mode'] = mode
+    nhs_funded = build_code_label_entry(
+        raw_course_data, lookup.nhs_funded, 'NHS')
+    if nhs_funded:
+        course['nhs_funded'] = nhs_funded
+    qualification = build_qualification(raw_course_data, kisaims)
+    if qualification:
+        course['qualification'] = qualification
+    sandwich_year = build_code_label_entry(
+        raw_course_data, lookup.sandwich_year, 'SANDWICH')
+    if sandwich_year:
+        course['sandwich_year'] = sandwich_year
+    title = build_eng_welsh_item('TITLE', raw_course_data)
+    if title:
+        course['title'] = title
+    if 'UCASPROGID' in raw_course_data:
+        course['ucas_code_id'] = raw_course_data['UCASPROGID']
```

Comment:
This field should be the `ucas_programme_id` that is stored in cosmos db. Sorry it's in the updated documentation, but it is probably worth changing it now. Likely to be overlooked later

Reply:
I agree have updated.